esadapter/src/main/java/com/ncc/neon/adapters/es/ElasticsearchAdapter.java

```diff
@@ -91,21 +94,44 @@ public HttpAsyncClientBuilder customizeHttpClient(HttpAsyncClientBuilder httpCli
         request.scroll(scroll);
         SearchResponse response = null;
         TabularQueryResult results = null;
-        List<Map<String, Object>> scrolledResults = null;
+        List<Map<String, Object>> collectedResults = null;
 
         try {
-            response = this.client.search(request, RequestOptions.DEFAULT);
-            if (query.getLimitClause() != null && query.getLimitClause().getLimit() > ES_BATCH_LIMIT) {
-                scrolledResults = ElasticsearchResultsConverter.getScrolledResults(scroll, response, this.client);
+            if (query.getLimitClause() != null && query.getLimitClause().getLimit() > ES_BATCH_LIMIT
+                    && query.getAggregateClauses() != null && !query.getAggregateClauses().isEmpty()) {
+                int numPartitions = query.getLimitClause().getLimit() / ES_BATCH_LIMIT;
+
+                TermsAggregationBuilder termsAB = null;
```

Comment:
So when I try increasing the limit on the timeline of the earthquakes dashboard to a big number (like 33,000) it just immediately returns no data (I think it's failing).  I'm wondering if that's because the date aggregations use a `DateHistogramAggregationBuilder`, and not a `TermsAggregationBuilder`.  You may want to add some checks on the type of `AggregationBuilder` that's in the search `request`.

Reply:
Interestingly, this seems to work fine with avg/max/min/sum aggregations, which use the `StatsAggregationBuilder`.  Maybe those requests also contain instances of `TermsAggregationBuilder`?  Whatever; as long as it works.